vendor/github.com/grafana/dskit/gate/gate.go

```diff
@@ -0,0 +1,161 @@
+package gate
```

Comment:
The code in this file should be removed from Mimir and we should use the one from dskit instead.